presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockBuilder.java

```diff
@@ -224,6 +225,7 @@ private void entryAdded(boolean isNull)
         }
         offsets[positionCount + 1] = values.getPositionCount();
         valueIsNull[positionCount] = isNull;
+        hasNullValue |= isNull;
```

Comment:
What about not growing `valueIsNull`?

Reply:
You mean to make `valueIsNull` field nullable in the builder? That would work too.
@dain do you have a preference?

Reply:
My preference is for consistency with with other block builders and it looks like they all use this style.

Reply:
Yet if there are not nulls, then we shouldn't allocate and grow `valueIsNull` at all.

Reply:
That seems like a good assumption, but I'd like to be more confident that it really matters before adding the complexity.